docs/src/main/sphinx/connector/iceberg.rst

```diff
@@ -510,15 +510,42 @@ Row level deletion
 Tables using v2 of the Iceberg specification support deletion of individual rows
 by writing position delete files.
 
-Rolling back to a previous snapshot
------------------------------------
+Snapshots
+---------
 
 Iceberg supports a "snapshot" model of data, where table snapshots are
-identified by an snapshot IDs.
+identified by a snapshot ID.
+
+The connector provides a system table containing snapshot information for every
+Iceberg table. Snapshots are identified by ``BIGINT`` snapshot IDs.
+You can find the snapshot IDs for table ``customer_orders`` by running the
+following command::
+
+    SELECT snapshot_id FROM iceberg.testdb."customer_orders$snapshots" ORDER BY committed_at DESC
+
+Time travel queries
+^^^^^^^^^^^^^^^^^^^
+
+The connector offers the ability to query historical data (i.e. the state of the table
+as it was in the previous day/week containing data which in the meantime has been
+modified or deleted) corresponding to a previous snapshot of an Iceberg table.
```

Comment:
```suggestion
The connector offers the ability to query historical data. This allows you to query the
table as it was when a snapshot of the table was made, even if the data has since been
modified or deleted.
```